### PR TITLE
CLI: Update Jetpack Package Generate Command

### PR DESCRIPTION
**Pull request by sdixon194**
#### Changes proposed in this Pull Request:

Updates the generate command to use a class based example.php file in the src folder, as well as add configuration for using $$next-version$$

Does the latter need to be something we ask the creator that they want, or should this be on by default? If by default, we can add the first 0.1.0-alpha version in the package src/class-example.php skeleton as well.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### Jetpack product discussion

https://github.com/Automattic/jetpack/pull/27906#discussion_r1047347016
#### Does this pull request change what data or activity we track or use?

No
#### Testing instructions:

* Use `jetpack generate packages --name test`. Things still work?
*

### CODE REVIEW

**Comment by sdixon194** [comment]
It might make sense to gate the versioning stuff behind if someone needs to create a mirror repo or not. Per this internal discussion: p1670959243165149-slack-CBG1CP4EN

> It depends on whether anyone is ever going to want to print the version somewhere.

I'm not sure in which context someone wouldn't want the versions printed somewhere.

**Comment by sdixon194** [comment]
> class-example.php:1:1: ERROR - Class file names should be based on the class name with "class-" prepended. Expected class-package-name.php, but found class-example.php. (WordPress.Files.FileName.InvalidClassFileName)

Do we need to maybe just ignore the entire skeleton directory in .phpcsignore, or did I not add it to `phpcs-excludelist.json` correctly?

**Comment by samiff** [comment]
> I'm not sure in which context someone wouldn't want the versions printed somewhere.

I don't think it hurts having the `extra.version-constants` added by default. It could always be removed if someone wanted, but not sure why they'd care 🤷 

> or did I not add it to phpcs-excludelist.json correctly?

Files specified there would have phpcs run against them only when new changes are made, so not quite what you were looking for. I think it makes sense to ignore the file more directly rather than gunking up the file with an inline ignore comment, so I specified it in `tools/cli/skeletons/packages/.phpcsignore` (we have support for per directory .phpcsignore).

File: tools/cli/skeletons/packages/src/class-example.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ *
+ * @package automattic/package-name
+ */
+
+namespace Automattic\Jetpack;
+
+/**
+ * Class description.
+ */
+class Package_Name {
```
Comment (samiff) [inline tools/cli/skeletons/packages/src/class-example.php:13]: The updated generate command works well enough for me. It'd be nice if `class Package_Name` was updated with the `package-name` as well, but not a dealbreaker.

File: tools/cli/commands/generate.js [hunk 2]
```diff
@@ -510,6 +511,9 @@ async function createComposerJson( composerJson, answers ) {
 			composerJson.extra[ 'branch-alias' ] = composerJson.extra[ 'branch-alias' ] || {};
 			composerJson.extra[ 'branch-alias' ][ 'dev-trunk' ] = '0.1.x-dev';
 			composerJson.extra.textdomain = name;
+			composerJson.extra[ 'version-constants' ] = {
+				'::PACKAGE_VERSION': `src/class-${ answers.name }.php`,
```
Comment (samiff) [inline tools/cli/commands/generate.js:520]: > If by default, we can add the first 0.1.0-alpha version in the package src/class-example.php skeleton as well.

Reminder if we want to get that added 👍

**Review by samiff — DISMISSED** [review]

File: tools/cli/skeletons/packages/src/class-example.php (hunk 1 above)
Comment (sdixon194) [reply tools/cli/skeletons/packages/src/class-example.php:13]: Nice, was gonna do this for a followup, but was easier than I thought.

File: tools/cli/commands/generate.js (hunk 2 above)
Comment (sdixon194) [reply tools/cli/commands/generate.js:520]: Added it for now, erring on the side that it's more useful to have it, and if someone doesn't they can just delete it. Still open for discussion thought.

**Comment by sdixon194** [comment]
Should be good for review again. Thanks for fixing the linter!

**Review by samiff — APPROVED** [review]

**Pull request merged**